### ISSUE TITLE
Case where learning_phase() is required when model does not have dropout

### ISSUE THREAD

**Issue by isaacgerg** [fchollet/keras#2304]
If you add a dropout layer and then remove it via model.layers, you still need to use K.learning_phase() even though there is no dropout layers in the model.  See the example code below that reproduces it.

https://gist.github.com/isaacgerg/5912700d93ffb6784400c95dc65d73bf

**Comment by fchollet** [comment]
Modifying layers does not change the underlying computational graph until you call `model.build()` (or `model.compile(optimizer, loss)`, which starts by calling `model.build()`).
